### PR TITLE
Dev agr stg

### PR DESCRIPTION
**Pull request by agr**
Adding a Nuget.Services.Storage nuget package containing the code from [Nuget.Services.Metadata/Catalog](https://github.com/NuGet/NuGet.Services.Metadata/tree/master/src/Catalog/Persistence) for easier reuse.
Code copied almost as is, the changes are:
* Namespace name;
* `Trace` calls were replaces with `ILogger` calls (and `ILoggerFactory` passed around).

No usable tests exist for the storage (or I was unable to find them), so nothing was copied.
Needed for implementing the [#3572](https://github.com/NuGet/NuGetGallery/issues/3572) and [#337](https://github.com/NuGet/Engineering/issues/337)

### CODE REVIEW

File: src/NuGet.Services.Storage/AzureStorageFactory.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.WindowsAzure.Storage;
+using System;
```
Comment (skofman1) [inline src/NuGet.Services.Storage/AzureStorageFactory.cs]: please order usings

Comment (agr) [inline src/NuGet.Services.Storage/AzureStorageFactory.cs]: Done

**Comment by joelverhagen** [comment]
> <?xml version="1.0" encoding="utf-8"?>

Can we use project.json please?

---
Refers to: src/NuGet.Services.Storage/packages.config:1 in 6b32119. [](commit_id = 6b32119119d8f0d7ee571d073453aa97a25c8fef, deletion_comment = False)

File: src/NuGet.Services.Storage/StorageListItem.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+using System;
+
+namespace NuGet.Services.Storage
+{
+    public class StorageListItem
+    {
+        public Uri Uri { get; set; }
+
+        public DateTime? LastModifiedUtc { get; set; }
```
Comment (joelverhagen) [inline src/NuGet.Services.Storage/StorageListItem.cs]: >set; [](start = 48, length = 4)

Can these be readonly?

File: src/NuGet.Services.Storage/project.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "dependencies": {
+    "WindowsAzure.Storage": "7.1.2",
```
Comment (joelverhagen) [inline src/NuGet.Services.Storage/project.json]: nice.

**Pull request merged**